custom_components/ica/caching.py

```diff
@@ -54,15 +86,30 @@ async def refresh(self):
         """Refreshes state using the value_factory"""
         # Invoke value factory (example: API)
         self._value = await self._value_factory()
+        self._timestamp = dt.datetime.now()
```

Comment:
Change all instances to use `utcnow()`

```suggestion
        self._timestamp = dt.datetime.utcnow()
```